pgmpy/factors/Factor.py

```diff
@@ -496,11 +496,24 @@ def __eq__(self, other):
         if type(self) != type(other):
             return False
 
-        if self.variables == other.variables and all(self.cardinality == other.cardinality) \
-                and all(self.values == other.values):
-            return True
-        else:
+        if set(self.scope()) != set(other.scope()):
             return False
+        else:
+            order_dict = {}
+            for var_index in range(len(self.scope())):
+                order_dict[var_index] = other.scope().index(self.scope()[var_index])
```

Comment:
@abinashpanda Suggest me something to make it more simpler. Assigning `self.scope()[var_index]` to a new variable and then using it doesn't make much sense to me.